exporter/elasticsearchexporter/internal/serializer/otelserializer/profile.go

```diff
@@ -37,50 +38,70 @@ func (*Serializer) SerializeProfile(resource pcommon.Resource, scope pcommon.Ins
 	if err != nil {
 		return err
 	}
-
-	for _, payload := range data {
-		event := payload.StackTraceEvent
-
-		if event.StackTraceID != "" {
-			if err = pushDataAsJSON(event, "", AllEventsIndex); err != nil {
-				return err
-			}
-			if err = serializeprofiles.IndexDownsampledEvent(event, pushDataAsJSON); err != nil {
-				return err
-			}
-		}
-
-		if payload.StackTrace.DocID != "" {
-			if err = pushDataAsJSON(payload.StackTrace, payload.StackTrace.DocID, StackTraceIndex); err != nil {
-				return err
-			}
-		}
-
-		for _, stackFrame := range payload.StackFrames {
-			if err = pushDataAsJSON(stackFrame, stackFrame.DocID, StackFrameIndex); err != nil {
-				return err
-			}
-		}
-
-		for _, executable := range payload.Executables {
-			if err = pushDataAsJSON(executable, executable.DocID, ExecutablesIndex); err != nil {
-				return err
-			}
-		}
-
-		for _, frame := range payload.UnsymbolizedLeafFrames {
-			if err = pushDataAsJSON(frame, frame.DocID, LeafFramesSymQueueIndex); err != nil {
-				return err
-			}
-		}
-
-		for _, executable := range payload.UnsymbolizedExecutables {
-			if err = pushDataAsJSON(executable, executable.DocID, ExecutablesSymQueueIndex); err != nil {
-				return err
-			}
-		}
-	}
-	return nil
+	return s.knownTraces.WithLock(func(tracesSet lru.LockedLRUSet) error {
+		return s.knownFrames.WithLock(func(framesSet lru.LockedLRUSet) error {
+			return s.knownExecutables.WithLock(func(executablesSet lru.LockedLRUSet) error {
+				return s.knownUnsymbolizedFrames.WithLock(func(unsymbolizedFramesSet lru.LockedLRUSet) error {
+					return s.knownUnsymbolizedExecutables.WithLock(func(unsymbolizedExecutablesSet lru.LockedLRUSet) error {
```

Comment:
My concern is that this code unnecessarily locks all the caches for the same time - until all data has been serialized.
So it possible makes sense to either split this up, which means the loop over `data` has to run 5 times (which possibly doesn't perform as bad as it sounds). Or think about using a single lock for all caches.

Reply:
I've added a benchmark and made the change. The difference (with a single profile) is minimal.

Before:
```
goos: darwin
goarch: arm64
pkg: github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/serializer/otelserializer
cpu: Apple M1 Max
BenchmarkSerializeProfile
BenchmarkSerializeProfile-10              337646              3520 ns/op           13110 B/op         51 allocs/op
PASS
```

After:
```
goos: darwin
goarch: arm64
pkg: github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/serializer/otelserializer
cpu: Apple M1 Max
BenchmarkSerializeProfile
BenchmarkSerializeProfile-10              285056              3652 ns/op           13110 B/op         51 allocs/op
```